### DISCUSSION TITLE
Docker file for AWS Python Deployement

### DISCUSSION THREAD

**Discussion by faisalriazgoraya** [seleniumbase/SeleniumBase#2378] [Q&A] [answered]
Facing issues while deploying python code using docker file. Can you please provide the complete docker file to AWS lambda deployement?

**Comment by mdmintz** [comment, marked as answer]
Sample Dockerfile: https://github.com/seleniumbase/SeleniumBase/blob/master/Dockerfile

**Reply by faisalriazgoraya** [reply]
My Docker File

# SeleniumBase Docker Image
FROM ubuntu:18.04

#=======================================
# Install Python and Basic Python Tools
#=======================================
RUN apt-get -o Acquire::Check-Valid-Until=false -o Acquire::Check-Date=false update
RUN apt-get install -y python3 python3-pip python3-setuptools python3-dev python-distribute
RUN alias python=python3
RUN echo "alias python=python3" >> ~/.bashrc

#=================================
# Install Bash Command Line Tools
#=================================
RUN apt-get -qy --no-install-recommends install \
    sudo \
    unzip \
    wget \
    curl \
    libxi6 \
    libgconf-2-4 \
    vim \
    xvfb \
  && rm -rf /var/lib/apt/lists/*

#================
# Install Chrome
#================
RUN curl -sS -o - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - && \
    echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list && \
    apt-get -yqq update && \
    apt-get -yqq install google-chrome-stable && \
    rm -rf /var/lib/apt/lists/*

#===========================
# Configure Virtual Display
#===========================
RUN set -e
RUN echo "Starting X virtual framebuffer (Xvfb) in background..."
RUN Xvfb -ac :99 -screen 0 1280x1024x16 > /dev/null 2>&1 &
RUN export DISPLAY=:99
RUN exec "$@"

#=======================
# Update Python Version
#=======================
RUN apt-get update -y
RUN apt-get -qy --no-install-recommends install python3.8
RUN rm /usr/bin/python3
RUN ln -s python3.8 /usr/bin/python3

#=============================================
# Allow Special Characters in Python Programs
#=============================================
RUN export PYTHONIOENCODING=utf8
RUN echo "export PYTHONIOENCODING=utf8" >> ~/.bashrc

#=====================
# Download WebDrivers
#=====================
RUN wget https://github.com/mozilla/geckodriver/releases/download/v0.33.0/geckodriver-v0.33.0-linux64.tar.gz
RUN tar -xvzf geckodriver-v0.33.0-linux64.tar.gz
RUN chmod +x geckodriver
RUN mv geckodriver /usr/local/bin/
RUN wget https://chromedriver.storage.googleapis.com/72.0.3626.69/chromedriver_linux64.zip
RUN unzip chromedriver_linux64.zip
RUN chmod +x chromedriver
RUN mv chromedriver /usr/local/bin/

# # Set the working directory inside the container
# WORKDIR /app

# Install Python dependencies
COPY requirements.txt .
RUN pip3 install -r requirements.txt

# Copy your SeleniumBase scripts or application code
COPY main2.py ./

# Set the entry point (replace 'your_script.py' with your actual script)
CMD [ "python3", "main2.py" ]

**Reply by faisalriazgoraya** [reply]
My Python Code
import time
from seleniumbase import Driver

proxy = "uqsqhklf:u4c1q2xhh3qd@38.154.227.167:5868"
user_agent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) ' \
             'Chrome/96.0.4664.45 Safari/537.36 '
driver = Driver(uc=True, agent=user_agent, headless=True)
driver.open("https://www.ibm.com/topics/natural-language-processing")
driver.sleep(30)
driver.save_screenshot('/tmp/screenshots/screenshot_{}.png'.format(int(time.time())))
driver.quit()

**Reply by faisalriazgoraya** [reply]
Facing this error using above code and docker file for AWS

INIT_REPORT Init Duration: 10134.04 ms	Phase: init	Status: timeout
--
Warning: uc_driver not found. Getting it now:
*** chromedriver to download = 120.0.6099.109 (Latest Stable)
Downloading chromedriver-linux64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/120.0.6099.109/linux64/chromedriver-linux64.zip ...
Cache folder (/home/sbx_user1051/.cache/selenium) cannot be created: Read-only file system (os error 30)
The chromedriver version (72.0.3626.69) detected in PATH at /usr/local/bin/chromedriver might not be compatible with the detected chrome version (120.0.6099.129); currently, chromedriver 120.0.6099.109 is recommended for chrome 120.*, so it is advised to delete the driver in PATH and retry
Traceback (most recent call last):
File "main2.py", line 7, in <module>
driver = Driver(uc=True, agent=user_agent, headless=True)
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/plugins/driver_manager.py", line 463, in Driver
driver = browser_launcher.get_driver(
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/core/browser_launcher.py", line 1562, in get_driver
return get_local_driver(
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/core/browser_launcher.py", line 3210, in get_local_driver
with uc_lock:  # Avoid multithreaded issues
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 166, in __enter__
gotten = self.acquire()
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 144, in acquire
self._do_open()
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 106, in _do_open
made_basedir = _ensure_tree(basedir)
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 43, in _ensure_tree
os.makedirs(path)
File "/usr/lib/python3.8/os.py", line 221, in makedirs
mkdir(name, mode)
OSError: [Errno 30] Read-only file system: b'downloaded_files'
INIT_REPORT Init Duration: 22894.84 ms	Phase: invoke	Status: error	Error Type: Runtime.ExitError
START RequestId: 6a36a7a0-c2d9-4341-8f8f-9f1d1ba3d50f Version: $LATEST
RequestId: 6a36a7a0-c2d9-4341-8f8f-9f1d1ba3d50f Error: Runtime exited with error: exit status 1Runtime.ExitError
END RequestId: 6a36a7a0-c2d9-4341-8f8f-9f1d1ba3d50f
REPORT RequestId: 6a36a7a0-c2d9-4341-8f8f-9f1d1ba3d50f	Duration: 22896.00 ms	Billed Duration: 22897 ms	Memory Size: 2048 MB	Max Memory Used: 109 MB
Warning: uc_driver not found. Getting it now:
*** chromedriver to download = 120.0.6099.109 (Latest Stable)
Downloading chromedriver-linux64.zip from:
https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/120.0.6099.109/linux64/chromedriver-linux64.zip ...
Cache folder (/home/sbx_user1051/.cache/selenium) cannot be created: Read-only file system (os error 30)
The chromedriver version (72.0.3626.69) detected in PATH at /usr/local/bin/chromedriver might not be compatible with the detected chrome version (120.0.6099.129); currently, chromedriver 120.0.6099.109 is recommended for chrome 120.*, so it is advised to delete the driver in PATH and retry
Traceback (most recent call last):
File "main2.py", line 7, in <module>
driver = Driver(uc=True, agent=user_agent, headless=True)
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/plugins/driver_manager.py", line 463, in Driver
driver = browser_launcher.get_driver(
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/core/browser_launcher.py", line 1562, in get_driver
return get_local_driver(
File "/usr/local/lib/python3.8/dist-packages/seleniumbase/core/browser_launcher.py", line 3210, in get_local_driver
with uc_lock:  # Avoid multithreaded issues
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 166, in __enter__
gotten = self.acquire()
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 144, in acquire
self._do_open()
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 106, in _do_open
made_basedir = _ensure_tree(basedir)
File "/usr/local/lib/python3.8/dist-packages/fasteners/process_lock.py", line 43, in _ensure_tree
os.makedirs(path)
File "/usr/lib/python3.8/os.py", line 221, in makedirs
mkdir(name, mode)
OSError: [Errno 30] Read-only file system: b'downloaded_files'
INIT_REPORT Init Duration: 1027.74 ms	Phase: init	Status: error	Error Type: Runtime.ExitError

**Reply by faisalriazgoraya** [reply]
Can you please help me in fixing this issue?

**Reply by faisalriazgoraya** [reply]
Thanks in Advance
